src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const arrLength = this.length;
+    let joinStr = '';
+
+    for (let i = 0; i < arrLength; i++) {
+      if (typeof this[i] === 'undefined' || this[i] === null) {
+        this[i] = '';
+      }
```

Comment:
here you can write the loop whithout if-else, you just can start from if this[i] is NOT undefined or null :)

Reply:
Agree, why don't you just add this[i] to string if it's not null or undefined, you're repeating yourself in next if block - try to invert your if statement to do that. Also, you don't need typeof here.